BitwardenShared/UI/Auth/Extensions/AuthRouter+Redirects.swift

```diff
@@ -354,13 +354,18 @@ extension AuthRouter {
                     return .landingSoftLoggedOut(email: activeAccount.profile.email)
                 }
 
-                // Otherwise, return `.vaultUnlock`.
-                return .vaultUnlock(
-                    activeAccount,
-                    animated: animated,
-                    attemptAutomaticBiometricUnlock: attemptAutomaticBiometricUnlock,
-                    didSwitchAccountAutomatically: didSwitchAccountAutomatically
-                )
+                let biometricUnlockStatus = try await services.biometricsRepository.getBiometricUnlockStatus()
+
+                if case let .available(_, enabled, validIntegrity) = biometricUnlockStatus, enabled, !validIntegrity {
+                    return .enterpriseSingleSignOn(email: activeAccount.profile.email)
```

Comment:
I think one other check we need to add here is to ensure the user doesn't have a master password, since otherwise they could use their MP to unlock the vault.
```
activeAccount.profile.userDecryptionOptions?.hasMasterPassword
```